File: scripts/algorithms/F/Find Missing Observations/Find Missing Observations.java
// Runtime: 3 ms (Top 98.11%) | Memory: 61.90 MB (Top 20.33%)

class Solution {
    public int[] missingRolls(int[] rolls, int mean, int n) {
        
    	int i,m=rolls.length,sum=0;
    	for(i=0;i<m;i++)
    		sum+=rolls[i];
			
    	int x=(mean*(m+n))-sum;
		
		if(x<=0||n*6<x||((x/n)==0)) {
    		return new int[] {};
		}
			
    	int arr[]=new int[n];
    	
		int p=x/n,q=x%n;
    	for(i=0;i<n;i++)  {
		
    		arr[i]=p+(q>0?1:0);
    		q--;
    	}
    	return arr;
    }
}

File: scripts/algorithms/F/Find Missing Observations/Find Missing Observations.js
// Runtime: 262 ms (Top 83.78%) | Memory: 64.50 MB (Top 83.78%)

var missingRolls = function(rolls, mean, n) {
  const diff = mean * (rolls.length + n) - rolls.reduce((a, b) => a + b);
  if (diff < n) return []; // Cannot be covered even by case 1 1 1 ...
  if (diff > 6 * n) return []; // Cannot be covered even by case 6 6 6 ...
  return Array(n)
    .fill(Math.ceil(diff / n), 0, diff % n)
    .fill(Math.floor(diff / n), diff % n);
};
